File: 0066-plus-one/0066-plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        if digits[-1] < 9:
            digits[-1] += 1
            return digits
        else:
            digits = digits[::-1]
            i, carry = 0, 1
            while i < len(digits):
                if digits[i]+carry > 9:
                    digits[i] = 0
                    i += 1
                else:
                    digits[i] += carry
                    carry = 0
                    break
            if carry == 1:
                digits.append(1)
            return digits[::-1]

File: 0066-plus-one/0066-plus-one.java
class Solution {
  public int[] plusOne(int[] digits) {
    if (digits[digits.length - 1] < 9) {
      digits[digits.length - 1]++;
      return digits;
    } else {
      List<Integer> dList = new ArrayList<Integer>();
      for (int n : digits) {
        dList.add(n);
      }
      Collections.reverse(dList);
      int i = 0;
      int carry = 1;
      while (i < dList.size()) {
        int value = dList.get(i) + carry;
        if (value > 9) {
          dList.set(i, 0);
          carry = 1;
          i++;
        } else {
          dList.set(i, value);
          carry = 0;
          break;
        }
      }
      if (carry == 1) dList.add(carry);
      Collections.reverse(dList);
      int[] ans = dList.stream().mapToInt(j->j).toArray();
      return ans;
    }
  }
}